Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -141,6 +146,67 @@ extension BrowserViewController: WKNavigationDelegate {
             decisionHandler(.cancel)
             return
         }
+        
+        if isUpholdOAuthAuthorization(url) {
+            decisionHandler(.cancel)
+            guard let queryItems = URLComponents(url: url, resolvingAgainstBaseURL: false)?.queryItems else {
+                return
+            }
+            var items: [String: String] = [:]
+            for query in queryItems {
+                items[query.name] = query.value
+            }
+            rewards.ledger.authorizeExternalWallet(
+                ofType: .uphold,
+                queryItems: items) { result, redirectURL in
+                    switch result {
```

Comment:
Thoughts on moving this code outside of wk nav delegate? 
I have a feeling that we expose too much details here, all these popup behaviors, hardcoded uphold url

We would need to pass webview and tab manager to new helper structure though, so it's not ideal

Reply:
I'll see if I can move the majority of it to `BVC+Rewards`